### PR TITLE
Extend FastFilter to support multiple properties...

### PR DESCRIPTION
**Pull request by KevinH-MS**
## Description
Previously, we would only use the "fast" filter if the filter expression used a single property name in all sub-expressions.  To improve Live Unit Testing execution performance, we need to be able to support an expression like this:
```
    FQN!=Test1&FQN!=Test2&TestCategory!=SkipWhenLiveUnitTesting
```
This change allows for use of multiple property names with the ```FastFilter```.  In cases where a single property is used, the time complexity remains O(n).  For cases using multiple properties, the worst case is O(m*n), but this is still better than the "slow" case, and in general, I expect "m" (number of property names used in the filter) to be small.

## Related issue
https://github.com/dotnet/testimpact/pull/2557

### CODE REVIEW

**Comment by KevinH-MS** [comment]
FYI @genlu @mayankbansal018 @abhishkk 

/CC @Microsoft/testimpact

File: test/Microsoft.TestPlatform.Common.UnitTests/Filtering/FastFilterTests.cs [hunk 1]
```diff
@@ -321,13 +313,74 @@ public void FastFilterShouldThrowExceptionForUnsupportedOperatorOperationCombina
...
             }
             catch (Exception ex)
             {
                 Assert.IsTrue(ex is ArgumentException);
                 Assert.AreEqual("An error occured while creating Fast filter.", ex.Message);
             }
         }
+
+        [TestMethod]
+        public void MultiplePropertyNames()
```
Comment (genlu) [inline test/Microsoft.TestPlatform.Common.UnitTests/Filtering/FastFilterTests.cs]: Could you also add a test for the other pattern? e.g. "FQN!=Test1&TestCategory!=SkipWhenLiveUnitTesting"

**Review by genlu — APPROVED** [review]
:shipit:

File: test/Microsoft.TestPlatform.Common.UnitTests/Filtering/FastFilterTests.cs (hunk 1 above)
Comment (KevinH-MS) [reply test/Microsoft.TestPlatform.Common.UnitTests/Filtering/FastFilterTests.cs]: Will do?

**Comment by KevinH-MS** [comment]
@cltshivash @mayankbansal018 @abhishkk would one of you be able to review/merge this if you approve?  We'd like to get this in for 15.7, as it allows for significant improvement in the Live Unit Testing case (it's required as part of the work to avoid sending the extraneous Discovery request in dotnet/testimpact#2557)

/CC @ManishJayaswal

**Review by mayankbansal018 — APPROVED** [review]
:shipit:

**Pull request merged**